pandas/core/groupby.py

```diff
@@ -2586,10 +2586,21 @@ def _get_grouper(obj, key=None, axis=0, level=None, sort=True,
     """
     group_axis = obj._get_axis(axis)
 
-    # validate that the passed level is compatible with the passed
+    # validate that the passed single level is compatible with the passed
     # axis of the object
     if level is not None:
-        if not isinstance(group_axis, MultiIndex):
+        # TODO: These two conditions are almost same.
+        # We should combine two.
+        if isinstance(group_axis, MultiIndex):
+            if is_list_like(level) and len(level) == 1:
```

Comment:
actually this condition I think you can pull out of the MultiIndex check here (as the else is the same condition)

Reply:
@jreback I am aware of this, but it seems that there are some processes only for non-`MultiIndex` condition in `else`. We have to consider carefully whether these are applicable for `MultiIndex`.
https://github.com/pandas-dev/pandas/pull/17621/files/e4cdd0726e685b0216056ba224ed363bf1e836f9#diff-720d374f1a709d0075a1f0a02445cd65R2618

Reply:
When these are applicable, we also have to check if there is no side effect to subsequent processes.